examples/custom_entry_buffer/entry_buffer/imp.rs

```diff
@@ -0,0 +1,128 @@
+use glib::{ParamFlags, ParamSpec, Value};
+use gtk::glib;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use once_cell::sync::Lazy;
+use std::cell::{Cell, RefCell};
+use std::rc::Rc;
+
+#[derive(Debug)]
+pub struct CustomEntryBuffer {
+    text: Rc<RefCell<String>>,
+    length: Cell<u16>,
+    max_length: Cell<u16>,
+}
+
+impl Default for CustomEntryBuffer {
+    fn default() -> Self {
+        Self {
+            text: Rc::new(RefCell::new(String::new())),
+            length: Cell::new(0),
+            max_length: Cell::new(0),
+        }
+    }
+}
+
+#[glib::object_subclass]
+impl ObjectSubclass for CustomEntryBuffer {
+    const NAME: &'static str = "CustomEntryBuffer";
+    type Type = super::CustomEntryBuffer;
+    type ParentType = gtk::EntryBuffer;
+}
+
+impl ObjectImpl for CustomEntryBuffer {
+    fn properties() -> &'static [ParamSpec] {
+        static PROPERTIES: Lazy<Vec<ParamSpec>> = Lazy::new(|| {
+            vec![
+                ParamSpec::new_uint(
+                    "length",
+                    "length",
+                    "Length",
+                    0,
+                    u16::MAX as u32,
+                    0,
+                    ParamFlags::READABLE,
+                ),
+                ParamSpec::new_uint(
+                    "max-length",
+                    "Maximum length",
+                    "Maximum number of characters for this entry",
+                    0,
+                    u16::MAX as u32,
+                    0,
+                    ParamFlags::READWRITE | ParamFlags::EXPLICIT_NOTIFY,
+                ),
+                ParamSpec::new_string(
+                    "text",
+                    "Text",
+                    "The contents of the buffer",
+                    Some(""),
+                    ParamFlags::READWRITE,
+                ),
+            ]
+        });
+        PROPERTIES.as_ref()
+    }
+
+    fn property(&self, _obj: &Self::Type, _id: usize, pspec: &ParamSpec) -> Value {
+        match pspec.name() {
+            "length" => (self.length.get() as u32).to_value(),
+            "max-length" => (self.max_length.get() as u32).to_value(),
+            "text" => self.text.borrow().to_value(),
+            _ => unreachable!(),
+        }
+    }
+
+    fn set_property(&self, _obj: &Self::Type, _id: usize, value: &Value, pspec: &ParamSpec) {
+        match pspec.name() {
+            "length" => {
+                self.length.set(value.get::<u32>().unwrap() as u16);
+            }
+            "max-length" => {
+                self.max_length.set(value.get::<u32>().unwrap() as u16);
+            }
+            "text" => {
+                self.text.replace(value.get().unwrap());
+            }
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl EntryBufferImpl for CustomEntryBuffer {
+    fn text(&self, _entry_buffer: &Self::Type) -> glib::GString {
+        self.text.borrow().clone().into()
```

Comment:
This is all not really great to use, is there a better way to handle this than having to clone the string? or is it fine to do so as the string can't be that huge due to the EntryBuffer limitation of being at max of a `u16::MAX` length?

Reply:
I don't know. You could in theory just return a reference here if it isn't required to be NUL-terminated.